t/plugin/kafka-logger.t

```diff
@@ -708,3 +708,52 @@ hello world
 --- error_log eval
 qr/send data to kafka: \{.*"body":"abcdef"/
 --- wait: 2
+
+
+
+=== TEST 24: set route(batch_max_size = 2), check if prometheus is initialized properly
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "kafka-logger": {
+                                "broker_list" :
+                                  {
+                                    "127.0.0.1":9092
+                                  },
+                                "kafka_topic" : "test2",
+                                "key" : "key1",
+                                "timeout" : 1,
+                                "batch_max_size": 2
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello"
+                }]]
+                )
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 25: access
```

Comment:
The test can pass even if I remove the change in the apisix/utils/batch-processor.lua.

Reply:
Can you show us the route information you configured?

Reply:
It means to delete your change, the newly added test case can also pass.

Reply:
I can not reproduce the problem, but the test case can reach the code line where the user report, the test case is not a good   one, i will try to reproduce again.

Reply:
I have reproduce the problem,  request the router with `prometheus` plugin disabled will cause the problem